src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -87,40 +88,109 @@ public static VariantContextWriter createVCFWriter(
+
         // optimization: if no input genotypes, just exit
         if (vc.getGenotypes().isEmpty()) return GenotypesContext.create();
 
-        // we need to determine which of the alternate alleles (and hence the likelihoods) to use and carry forward
-        final List<Integer> likelihoodIndexesToUse = determineLikelihoodIndexesToUse(vc, allelesToUse);
+        // find the likelihoods indexes to use from the used alternate alleles
+        final List<Integer> sacIndexesToUse = determineSACIndexesToUse(vc, allelesToUse);
 
         // create the new genotypes
-        return createGenotypesWithSubsettedLikelihoods(vc.getGenotypes(), vc, allelesToUse, likelihoodIndexesToUse, assignGenotypes);
+        return createGenotypesWithSubsettedLikelihoods(vc.getGenotypes(), vc, allelesToUse, likelihoodIndexesToUse, sacIndexesToUse, assignGenotypes);
+    }
+
+
+    /**
+     * Find the likelihood indexes to use for a selected set of diploid alleles
+     *
+     * @param originalVC        the original VariantContext
+     * @param allelesToUse      the subset of alleles to use
+     * @return a list of PL indexes to use or null if none
+     */
+    private static List<Integer> determineDiploidLikelihoodIndexesToUse(final VariantContext originalVC, final List<Allele> allelesToUse) {
+
+        Utils.nonNull(originalVC, "VariantContext is null");
+        Utils.nonNull(allelesToUse, "allelesToUse is null");
+
+        // the bitset representing the allele indexes we want to keep
+        final boolean[] alleleIndexesToUse = getAlleleIndexBitset(originalVC, allelesToUse);
+
+        // an optimization: if we are supposed to use all (or none in the case of a ref call) of the alleles,
+        // then we can keep the PLs as is; otherwise, we determine which ones to keep
+        if ( MathUtils.countOccurrences(true, alleleIndexesToUse) == alleleIndexesToUse.length )
+            return null;
+
+        return getDiploidLikelihoodIndexes(originalVC, alleleIndexesToUse);
     }
 
     /**
-     * Create the new GenotypesContext with the subsetted PLs and ADs
+     * Get the actual likelihoods indexes to use given the corresponding diploid allele indexes
+     *
+     * @param originalVC           the original VariantContext
+     * @param alleleIndexesToUse   the bitset representing the alleles to use (@see #getAlleleIndexBitset)
+     * @return a non-null List
+     * @throws GATKException if the original VariantContext contains a genotype that is not dipliid
+     */
+    private static List<Integer> getDiploidLikelihoodIndexes(final VariantContext originalVC, final boolean[] alleleIndexesToUse) {
+
+        Utils.nonNull(originalVC, "VariantContext is null");
+        Utils.nonNull(alleleIndexesToUse, "alleleIndexesToUse is null");
+
+        // All samples must be diploid
+        for ( final Genotype g : originalVC.getGenotypes() ){
+            if ( g.getPloidy() != DEFAULT_PLOIDY )
```

Comment:
Note that `public static final int DEFAULT_PLOIDY = HomoSapiensConstants.DEFAULT_PLOIDY`. Would it be clearer if I just used `HomoSapiensConstants.DEFAULT_PLOIDY` directly?